experiments/xls/stream_logging.py

```diff
@@ -21,4 +23,44 @@ def __exit__(self, exc_type, exc_value, traceback):
         xls_logger.removeHandler(self.handler)
 
     def get_log(self) -> str:
-        return self._stream.getvalue()
\ No newline at end of file
+        return self._stream.getvalue()
+
+
+class LogParser(object):
+    """
+    Current logging records are generated as strings with the following
+    format "LEVEL::MESSAGE". This parser takes the raw log strings and turns
+    them into a dictionary for ease of access of the level and message props.
+    """
+
+    def __init__(self, raw_logs: [str] = []):
+        self.raw_logs = raw_logs
+        self.logs = []
+        self.parse_logs(self.raw_logs)
+
+    def parse_logs(self, raw_logs: [str]):
```

Comment:
regarding type annotations: you should also add the return type:
`def parse_logs(self, raw_logs: [str]) -> None:`